agent/checks/check_test.go

```diff
@@ -836,12 +836,19 @@ func TestCheckHTTP_TLS_BadVerify(t *testing.T) {
 		if got, want := notif.State(cid), api.HealthCritical; got != want {
 			r.Fatalf("got state %q want %q", got, want)
 		}
-		if !strings.Contains(notif.Output(cid), "certificate signed by unknown authority") {
+		if !isInvalidCertificateError(notif.Output(cid)) {
 			r.Fatalf("should fail with certificate error %v", notif.OutputMap())
 		}
 	})
 }
 
+// isInvalidCertificateError checks the error string for an untrusted certificate error.
+// The specific error message is different on Linux and macOS.
```

Comment:
Can we add a "this is safe to remove when https://github.com/golang/go/issues/52010 is addressed"

Reply:
👍 https://github.com/hashicorp/consul/pull/15903/commits/f7c028a4c0ee27a6ad79d92ea01413c893c7acb2